reframe/core/schedulers/__init__.py

```diff
@@ -321,7 +327,8 @@ def wait(self):
         if self.jobid is None:
             raise JobNotStartedError('cannot wait an unstarted job')
 
-        return self.scheduler.wait(self)
+        self.scheduler.wait(self)
+        self._completion_time = datetime.now()
```

Comment:
You should set the completion time also when `finished()` is called and set it here only if it is not already set.